IPs/RSKIP491.md

```diff
@@ -0,0 +1,59 @@
+---
+rskip: 491
+title: Reduce target difficulty to lower average block time to 10s
+description: This RSKIP proposes a 30% reduction in target difficulty to lower the average block time from 14 seconds to 10 seconds, improving transaction finality while maintaining network security.
+status: Draft
+purpose: Usa
+author: PDG (@patogallaiovlabs)
+layer: Core
+complexity: 1
+created: 2025-02-27
+---
+
+# Reduce target difficulty to lower average block time to 10s
+
+|RSKIP          | 491        |
+| :------------ |:-------------|
+|**Title**      |Reduce target difficulty to lower average block time to 10s |
+|**Created**    |27-FEB-2025 |
+|**Author**     |Patricio Gallardo|
+|**Purpose**    |Usa|
+|**Layer**      |Core |
+|**Complexity** |1 |
+|**Status**     |Draft |
+
+## Abstract
+
+This RSKIP proposes reducing the target difficulty by 30% to achieve an average block time of 10 seconds instead of the current 14 seconds. By adjusting the difficulty calculation, Rootstock can improve transaction finality while maintaining network security and stability.
+
+## Motivation  
+Currently, Rootstock maintains a 14-second block time, balancing security and efficiency.  
+
+Recent analysis of block template refresh intervals (as explored in [this research](https://blog.rootstock.io/noticia/leveraging-bitcoins-security-exploring-the-dynamics-of-merged-mining/)) showed that several mining pools have improved their block template refresh policies, reducing the time it takes to update new templates. These improvements have led to:  
+- a **decrease in the sibling-to-main block ratio**,  
+- and **faster effective block times**, previously averaging 30 seconds and now around [24 seconds](https://stats.rootstock.io/).  
+
+With these optimizations in place, reducing the block time to **10 seconds** is now more feasible, as the network can sustain faster block production without compromising stability.  
+
+By leveraging these mining pool improvements, this proposal aims to:  
+- **Reduce block time to 10 seconds** with minimal impact on network congestion while maintaining stability.  
+- **Enhance transaction finality**, improving efficiency without compromising security.  
+- **Encourage continued improvements in mining pool behavior**, reinforcing the benefits of optimized template refresh strategies.  
+
+
+## Specification  
+To achieve a 10-second block time, the target block time reference must be changed from **14s to 10s** in the difficulty adjustment calculation.  
+
+In RSKj, this value is configured in the **`Constants`** class, which defines the target block time for mainnet. The relevant code can be found [here](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/config/Constants.java#L250).  
```

Comment:
is this only intented for mainnet or do we think it makes sense to replicate the adjustments to Testnet as well?

Reply:
it makes sense actually, would be a good experiment before mainnet.